ui/src/main/java/edu/wpi/grip/ui/pipeline/source/SourceViewFactory.java

```diff
@@ -0,0 +1,34 @@
+package edu.wpi.grip.ui.pipeline.source;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.Source;
+import edu.wpi.grip.core.sources.CameraSource;
+import edu.wpi.grip.core.sources.MultiImageFileSource;
+
+/**
+ * Factory for creating views to control sources.
+ */
+public final class SourceViewFactory {
+    private SourceViewFactory() { /* no op */ }
+
+    /**
+     * Create an instance of {@link SourceView} appropriate for the given socket.
+     *
+     * @param eventBus The EventBus
+     * @param source   The source to create the view for
+     * @param <S>      The type of the source
+     * @return The appropriate SourceView.
+     */
+    public static <S extends Source> SourceView<S> createSourceControlsView(EventBus eventBus, S source) {
+        if (source instanceof CameraSource) {
+            return (SourceView<S>) new CameraSourceView(eventBus, (CameraSource) source);
+        } else if (source instanceof MultiImageFileSource) {
+            return (SourceView<S>) new MultiImageFileSourceView(eventBus, (MultiImageFileSource) source);
+        } else {
+            final SourceView<S> newSourceView = new SourceView<S>(eventBus, source) {
+            };
+            eventBus.register(newSourceView);
```

Comment:
Why does this register `SourceView` but no other type of source view?  It seems inconsistent and will make it harder to refactor `eventBus.register` if we do it in various different places